assets/javascripts/discourse/initializers/preview-edits.js.es6

```diff
@@ -154,7 +164,17 @@ export default {
           }
       	},
 
-        applyMasonry(){
+        updateHideCategory() {
+          const mobile = this.get('site.mobileView');
+          if (!mobile && this.settingEnabled('topic_list_category_badge_move')) {
```

Comment:
this is done.  However, it's introduced a minor issue - badges are not hidden in Category view.  This is a compromise as I've not yet found a way to keep all the elements switching correctly without the current implementation I have.

Reply:
> this is done

:+1:

> hideCategory is being repurposed to switch the category column off and on, but you can't at the same time use it to switch category label off

Why not?

Reply:
OK let's see if I can explain it more clearly: as we don't have control over the Topic List template so that we can wire in additional logic, we have to make do with hideCategory in Topic List Header.  So, hideCategory has to be set when we are in a Category list or if we have NOT set showCategoryColumn in order to remove the table header entry for Category.  showCategoryColumn is a configurable site setting and merely states whether we move the badge to a Column.  If however we don't want a category column, showCategoryColumn will be false and the topic list item badge will show.   Given hideCategory now depends on the value showCategoryColumn, we can't use it to determine the value of showCategoryColumn.  If there is any other independent way to tell if we are on a Category view within the Topic List Item code we might be able to set a new third boolean, but we can't use hideCategory because otherwise we'd create a circular dependency.

Reply:
There must be a way. Why do we need the ``showCategoryColumn`` property? Can we just use ``hideCategory`` to determine whether we both show the category column and the category badge?